core/include/vecmem/utils/impl/copy.ipp

```diff
@@ -543,4 +678,216 @@ bool copy::is_contiguous(const data::vector_view<TYPE>* data,
     return true;
 }
 
+template <typename... VARTYPES, std::size_t INDEX, std::size_t... Is>
```

Comment:
NIT: I guess this recursion could be implemented without the full integer sequence by just decrementing.

Reply:
:thinking: Yeah... You're right, just a single `INDEX` template parameter could be enough. How strongly do you feel about this for the first version of the code?

Reply:
Since the templating could indeed be simplified for all of the implementation functions, I went ahead and did that.